pkg/poller/testutil.go

```diff
@@ -0,0 +1,108 @@
+package poller
+
+import (
+	"sync"
+)
+
+type TestCollector struct {
+	sync.RWMutex
+	Logger Logger
+	inputs []*InputPlugin
+	poller *Poller
+}
+
+var _ Collect = &TestCollector{}
```

Comment:
Not a fan of global variables. Necessary? Can this file live in a separate package, like `testutil`?

Reply:
It could this, is also just a compile time test to ensure it conforms to the interface. Not entirely necessary